src/ccall/src/goat.rs

```diff
@@ -49,6 +50,7 @@ pub struct Mesh<'a> {
     mesh: &'a MeshInternal,
 }
 
+// TODO why are buffer timelines allowed to outlive mesh timelines?
```

Comment:
@IRog  halp, I thought `'a` means lifetime of returned buffers should be the same as the Mesh object but that doesn't seem to be the case.

Reply:
I'm not 100% sure but I think the issue is related to the `unsafe` usage and the borrow checker not having all the information it needs.

I'll look more once I've fixed my stack install and can run the haskell side again...